src/Pos/Generator/Block/Logic.hs

```diff
@@ -55,35 +61,36 @@ genBlocks params = do
 -- Generate a valid 'Block' for the given epoch or slot (genesis block
 -- in the former case and main block the latter case) and apply it.
 genBlock ::
-       (MonadBlockGenBase m)
+       forall ctx m g. (RandomGen g, MonadBlockGen ctx m)
     => EpochOrSlot
-    -> BlockGenMode m (Blund SscGodTossing)
+    -> BlockGenRandMode g m (Blund SscGodTossing)
```

Comment:
are you sure it's necessary to reflect fact that generator is random in the name `BlockGenRandMode`? it seems to be more or less obvious